versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -725,6 +734,11 @@ protected final CommitLogEntry fetchFromCommitLog(OP_CONTEXT ctx, Hash hash) {
    * returned as {@code null}.
    */
   private List<CommitLogEntry> fetchMultipleFromCommitLog(OP_CONTEXT ctx, List<Hash> hashes) {
+    if (hashes.contains(NO_ANCESTOR)) {
+      // Do not try to fetch NO_ANCESTOR - it won't exist.
+      hashes = new ArrayList<>(hashes);
+      hashes.remove(NO_ANCESTOR);
```

Comment:
javadoc: `The returned list must have exactly as many elements as in the parameter {@code hashes}.`

Aren't we violating the javadoc contract of this method here?

Reply:
Duh! You're right!